onnxruntime/core/graph/contrib_ops/contrib_defs.cc

```diff
@@ -2843,6 +2843,59 @@ It's an extension of Gelu. It takes the sum of input A and bias input B as the i
         }
       });
 
+  static const char* Triu_ver1_doc = R"DOC(
+	  Returns the upper triangular part of a 2-D matrix, or batches of 2-D matrices. Triu takes one input tensor of shape
+	  [*, N, M], where * is zero or more batch dimensions. The upper triangular part consists of the elements on or
+	  above the given diagonal (k).
+	  If k = 0, this op retains elements on or above the the main diagonal. If k > 0, this op
+	  retains elements on or above k diagonals over the main diagonal. If k < 0, this op retains elements
+	  on or above as many diagonals below the main diagonal. All other elements in the matrix are set to zero.
+	  )DOC";
+
+  ONNX_CONTRIB_OPERATOR_SCHEMA(Triu)
+      .SetDomain(kMSDomain)
+      .SinceVersion(1)
+      .SetDoc(Triu_ver1_doc)
+      .SetSupportLevel(OpSchema::SupportType::EXPERIMENTAL)
+        .Input(
+            0,
+            "X",
+            "Input tensor of rank 2 or higher.",
+            "T1")
+        .Input(
+            1,
+            "k",
+            "A 0-D tensor containing a single value corresponding to the number diagonals above or the main diagonal to exclude or include.",
+            "tensor(int64)",
+            OpSchema::Optional)
+        .Output(
+            0,
+            "Y",
+            "Output tensor of the same type and shape as the input tensor.",
+            "T1")
+        .TypeConstraint(
+            "T1",
```

Comment:
This type constraint doesn't match what you are using in the kernel. I think you meant to use just "T" not "T1" here?